velox/functions/prestosql/aggregates/tests/MaxSizeForStatsTest.cpp

```diff
@@ -0,0 +1,261 @@
+// []
+// [1,2,3]
+// Query:
+// select aggr(c0)
+// Output
+// 44
+TEST_F(MaxSizeForStatsTest, arrayGlobalAggregate) {
+  auto vectors = {makeRowVector({makeArrayVector<int64_t>({
+      {1, 2, 3, 4, 5},
+      {},
+      {1, 2, 3},
+  })})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 44");
+}
+
+// Input:
+// c0(map(tinyint,integer))
+// map(array(1,1,1,1,1),array(1,1,1,1,1))
+// map()
+// map(array(1,1,1), array(1,1,1))
+// Query:
+// select aggr(c0)
+// Output
+// 29
+TEST_F(MaxSizeForStatsTest, mapGlobalAggregate) {
+  auto vectors = {makeRowVector({makeMapVector<int8_t, int32_t>(
+      {{{1, 1}, {1, 1}, {1, 1}, {1, 1}, {1, 1}},
+       {},
+       {{1, 1}, {1, 1}, {1, 1}}})})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 29");
+}
+
+// Input:
+// c0( row(array(bigint), map(tinyint,integer))
+// row(aray[1,2,3,4,5], map(array(1,1,1,1,1),array(1,1,1,1,1)))
+// row(array[], map())
+// row(array[1,2,3], map(array(1,1,1), array(1,1,1)))
+// Query:
+// select aggr(c0)
+// Output
+// 77
+TEST_F(MaxSizeForStatsTest, rowGlobalAggregate) {
+  auto vectors = {makeRowVector({makeRowVector(
+      {makeArrayVector<int64_t>({
+           {1, 2, 3, 4, 5},
+           {},
+           {1, 2, 3},
+       }),
+       makeMapVector<int8_t, int32_t>(
+           {{{1, 1}, {1, 1}, {1, 1}, {1, 1}, {1, 1}},
+            {},
+            {{1, 1}, {1, 1}, {1, 1}}})})})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 77");
+}
+
+// Input:
+// c0(varbinary)
+// "buf"
+// ""
+// "hello"
+// query:
+// select aggr(c0)
+// output:
+// 9
+TEST_F(MaxSizeForStatsTest, varbinaryGlobalAggregate) {
+  VectorPtr varbinary_vector = BaseVector::create(VARBINARY(), 3, pool());
+  auto flat_vector = varbinary_vector->asFlatVector<StringView>();
+  StringView sv1{"buf"};
+  StringView sv2{""};
+  StringView sv3{"henlo"};
+  flat_vector->set(0, sv1);
+  flat_vector->set(1, sv2);
+  flat_vector->set(2, sv3);
+
+  auto vectors = {makeRowVector({varbinary_vector})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 9");
+}
+
+// Input:
+// c0(varchar)
+// "{1, 2, 3, 4, 5}"
+// "{}"
+// "{1, 2, 3}"
+// query:
+// select aggr(c0)
+// output:
+// 19
+TEST_F(MaxSizeForStatsTest, varcharGlobalAggregate) {
+  auto vectors = {makeRowVector({makeFlatVector<StringView>({
+      "{1, 2, 3, 4, 5}",
+      "{}",
+      "{1, 2, 3}",
+  })})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 19");
+}
+
+// Input (3 rows of recursive complex data):
+// c0(row(varchar, map(tinyint, bigint)))
+// ("{1, 2, 3, 4, 5}", {1:null,})
+// ("{}", {2:[4,5,null]})
+// ("{1, 2, 3}", {null:[7,8,9]})
+// query:
+// select aggr(c0)
+// output:
+// 50 (3rd row(4), 13("{1, 2, 3}") + 5 (null) + 28([7,8,9]))
+TEST_F(MaxSizeForStatsTest, complexRecursiveGlobalAggregate) {
```

Comment:
I don't see any tests that use Constant or Dictionary vectors as inputs. It would be nice to add these.

Reply:
done, added constantEncodingTest and dictionaryEncodingTest

Reply:
Thanks.